server/src/main/java/org/apache/iotdb/db/engine/StorageEngine.java

```diff
@@ -325,6 +325,7 @@ public StorageGroupProcessor getProcessor(PartialPath path) throws StorageEngine
       storageGroupPath = storageGroupMNode.getPartialPath();
       StorageGroupProcessor processor = processorMap.get(storageGroupPath);
       if (processor == null) {
+        waitAllSgReady(storageGroupPath);
```

Comment:
why have to wait for all SGs ready?

Reply:
Just for simplicity, we could optimize it later.

Reply:
Otherwise, it will throw an exception directly, failing current operations.